src/main/java/org/jenkins/plugins/lockableresources/queue/LockableResourcesStruct.java

```diff
@@ -35,11 +35,26 @@ public LockableResourcesStruct(RequiredResourcesProperty property, EnvVars env)
     required = new ArrayList<>();
 
     LockableResourcesManager resourcesManager = LockableResourcesManager.get();
+    List<LockableResource> declaredResources = resourcesManager.getDeclaredResources();
     for (String name : property.getResources()) {
       String resourceName = env.expand(name);
       if (resourceName == null) {
         continue;
       }
+
+      if (resourceName.contains("$")) {
```

Comment:
Scenario:
	- declared resource: `resource-1`
	- job configuration:
		- lock: `resource-$id`
		- build parameter: `id`
when triggering the job with `id=1`, the resource `resource-1` will be effectively locked and released after the execution finishes. But an ephemeral resource named `resource-$id` will be created and queued by the current build, when `LockableResourcesQueueTaskDispatcher` tests is the build can run (LockableResourcesQueueTaskDispatcher.java:108). No expansion of the variable `id` occurs at this point, because the EnvVars object is still empty. The ephemeral resource is eventually remove from the queue after a minute or so.

The solution I proposed is to skip creating ephemeral resources for names containing the dollar symbol `$` that haven't been previously declared (i.e. from Jenkins configuration manager).

Reply:
@TobiX do you have any comments on this?

Reply:
Since "queued" resources are an implementation detail, maybe just hide these from the UI (maybe only hide queued & ephemeral)? I'm not sure if we break something if we just skip the creation of those resources...

Reply:
I understand. Although if we hide it the user may not understand why the next build cannot run (i.e. there's an ephemeral resource queued but cannot know because is not shown in the ui)
I'll try to implement the changes somewhere during this week

Reply:
I reverted this change and left the UI as it is, without hiding the ephemeral & queued resources. I think is better to show that there's a queued ephemeral resource than having the user wondering why the next build trying to lock its resource cannot run yet.